web/src/helpers/nodes.ts

```diff
@@ -56,3 +58,62 @@ type SearchDelimiterMap = typeof searchDelimiterMap
 export function parseSearchGroup(nodeId: string, field: keyof SearchDelimiterMap) {
   return nodeId.split(':')[searchDelimiterMap[field]] || ''
 }
+
+export function stateTypeColor(state: RunState | EventType) {
+  switch (state) {
+    case 'NEW':
+      return theme.palette.secondary.main
+    case 'START':
+      return theme.palette.info.main
+    case 'RUNNING':
+      return theme.palette.info.main
+    case 'COMPLETE':
+      return theme.palette.primary.main
+    case 'COMPLETED':
+      return theme.palette.primary.main
+    case 'FAIL':
+      return theme.palette.error.main
+    case 'FAILED':
+      return theme.palette.error.main
+    case 'ABORT':
+      return theme.palette.warning.main
+    case 'ABORTED':
+      return theme.palette.warning.main
+    default:
+      return theme.palette.secondary.main
+  }
+}
+
+export function jobRunsStatus(runs: Run[], limit = -14) {
```

Comment:
`-14` is kind of a weird default parameter. Can we make this a positive integer and adjust the slice?

Reply:
Yeap, it's not the best practice and can confuse, updated